guides/hack/01-getting-started/03-starting-a-real-project.md

```diff
@@ -0,0 +1,332 @@
+Loading composer repositories with package information
+Updating dependencies (including require-dev)
+Package operations: 2 installs, 0 updates, 0 removals
+  - Installing hhvm/hsl (v4.0.0): Loading from cache
+  - Installing hhvm/hhvm-autoload (v2.0.3): Loading from cache
+Writing lock file
+Generating autoload files
+/var/folders/3l/2yk1tgkn7xdd76bs547d9j90fcbt87/T/tmp.xaQwE1xE/vendor/autoload.hack
+$ composer require --dev hhvm/hhast hhvm/hacktest facebook/fbexpect
+Using version ^4.0 for hhvm/hhast
+Using version ^1.4 for hhvm/hacktest
+Using version ^2.5 for facebook/fbexpect
+./composer.json has been updated
+Loading composer repositories with package information
+Updating dependencies (including require-dev)
+Package operations: 7 installs, 0 updates, 0 removals
+  - Installing facebook/difflib (v1.1): Loading from cache
+  - Installing hhvm/hsl-experimental (v4.0.1): Loading from cache
+  - Installing hhvm/type-assert (v3.3.1): Loading from cache
+  - Installing facebook/hh-clilib (v2.1.0): Loading from cache
+  - Installing hhvm/hhast (v4.0.4): Loading from cache
+  - Installing hhvm/hacktest (v1.4): Loading from cache
+  - Installing facebook/fbexpect (v2.5.1): Loading from cache
+Writing lock file
+Generating autoload files
+/private/var/folders/3l/2yk1tgkn7xdd76bs547d9j90fcbt87/T/tmp.xaQwE1xE/vendor/autoload.hack
+$
+```
+
+### Adding Functions or Classes
+
+As a toy example, we're going to create a function that squares a vector of
+numbers; save the following as `src/square_vec.hack`:
+
+```Hack
+use namespace HH\Lib\Vec;
+
+function square_vec<T as num>(vec<T> $numbers): vec<int> {
+  return Vec\map($numbers, $number ==> $number * $number);
+}
+```
+
+If you then run `hh_client`, it will tell you of a mistake:
+
+```
+src/square_vec.hack:4:10,57: Invalid return type (Typing[4110])
+  src/square_vec.hack:3:53,55: This is an int
+  src/square_vec.hack:4:40,56: It is incompatible with a num (int/float) because this is the result of an arithmetic operation with a num as the first argument, and no floats.
+  src/square_vec.hack:3:35,35: Here is why I think the argument is a num: this is a num
+```
+
+To fix this, change the return type of the fucntion from `vec<int>` to `vec<num>`.
+
+We now have a function that is valid Hack, but it's not tested, and nothing calls it.
+
+### Adding an Executable
+
+Save the following as `bin/square_some_things.hack`:
+
+```Hack
+#!/usr/bin/env hhvm
+
+require_once(__DIR__.'/../vendor/autoload.hack');
+
+<<__EntryPoint>>
+async function main(): Awaitable<noreturn> {
+  \Facebook\AutoloadMap\initialize();
+
+  $squared = square_vec(vec[1, 2, 3, 4, 5]);
+  foreach ($squared as $square) {
+    printf("%d\n", $square);
+  }
+  exit(0);
+}
+```
+
+This program:
+ - requires and initializes the autoloader so that the function we just defined
+   can be found
+ - calls the function
+ - prints the results
+ - exits
+
+ Unlike C, there is nothing special about the name `main`; the
```

Comment:
Where possible, I think our prose is clearer where we don't assume knowledge of specific other languages.

```suggestion
There is nothing special about the name `main`; the
```